oak_functions/loader/tests/integration_test.rs

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use lazy_static::lazy_static;
+use maplit::hashmap;
+use oak_functions_abi::proto::{Request, Response};
+use oak_functions_loader::{
+    logger::Logger,
+    lookup::{LookupData, LookupDataSource},
+    server::WasmHandler,
+};
+use std::{io::Write, sync::Arc};
+use tokio;
+
+lazy_static! {
+    static ref WASM_MODULE_BYTES: Vec<u8> = {
+        let mut manifest_path = std::env::current_dir().unwrap();
+        manifest_path.push("tests");
+        manifest_path.push("module");
+        manifest_path.push("Cargo.toml");
+
+        test_utils::compile_rust_wasm(manifest_path.to_str().unwrap(), false)
+            .expect("Could not read Wasm module")
+    };
+}
+
+#[tokio::test]
+async fn test_consistent_lookup() {
+    let entries = hashmap! {
+       b"ConsistentLookup".to_vec() => b"Value1".to_vec(),
+    };
+    let serialized_entries = test_utils::serialize_entries(entries);
+    let temp_file = tempfile::NamedTempFile::new().unwrap();
+    temp_file.as_file().write_all(&serialized_entries).unwrap();
+
+    let lookup_data = crate::LookupData::new_empty(
+        Some(LookupDataSource::File(temp_file.path().to_path_buf())),
+        Logger::for_test(),
+    );
+    lookup_data.refresh().await.unwrap();
+
+    let wasm_handler = WasmHandler::create(
+        &WASM_MODULE_BYTES,
+        Arc::new(lookup_data),
+        vec![],
+        Logger::for_test(),
+    )
+    .expect("Could not instantiate WasmHandler.");
+
+    let request = Request {
+        body: b"ConsistentLookup".to_vec(),
+    };
+
+    let response: Response = wasm_handler.handle_invoke(request).await.unwrap();
+    assert_eq!(response.body().unwrap(), b"Value1");
+
+    let entries = hashmap! {
+       b"ConsistentLookup".to_vec() => b"Value2".to_vec(),
+    };
+    let serialized_entries = test_utils::serialize_entries(entries);
+    temp_file.as_file().write_all(&serialized_entries).unwrap();
+
+    wasm_handler.refresh_lookup_data().await.unwrap();
```

Comment:
@project-oak/core Here I need to refresh the lookup data for the wasm_handler after changing the lookupdata in the external file. As `lookupdata` is not public in `WasmHandler`, I implemented `refresh_lookup_data` for WasmHandler. 

I am not particularily happy about it as I implement functionality only for testing. Do you see any other way (e.g., waiting for a automatic refresh/refactor this as a unit test) or do you think it is alright to add `refresh_lookup_data` for `WasmHandler`?

Reply:
I think you can change `lookup_data` declared in line 48 to be `Arc<LookupData>`. Then pass `lookup_data.clone()` to `WasmHandler::create` and call `lookup_data.refresh()` here instead of having to add a method to `WasmHandler`. Kind of similar to https://github.com/project-oak/oak/blob/6769de3ec5b2e3115708d45f9eb7b19528886c29/oak_functions/loader/src/main.rs#L271-L279

Reply:
That worked---thank you very much! :)